File: Ruby/labyrinth_character.rb
# frozen_string_literal: true
module Irrgarten

    class Labyrinth_character
      attr_reader :intelligence, :strength, :row, :col
      attr_accessor :health

    def initialize(name, intelligence, strength, health)
      @name = name
      @intelligence = intelligence
      @strength = strength
      @health = health
      @row = 0
      @col = 0
    end

    def self.copy(other)
      @name = other.name
      @intelligence = other.intelligence
      @strength = other.strength
      @health = other.health
      @row = other.row
      @col = other.col
    end

    def dead
      @health <= 0
    end

    def get_row
      @row
    end

    def get_col
      @col
    end

    def to_s
        "M Name: #{@name}, Intelligence: #{@intelligence}, Strength: #{@strength}, Health: #{@health}, Weapons: #{@weapons}, Shields: #{@shields}, Row: #{@row}, Col: #{@col}"
    end

    def attack
      raise NotImplementedError, "#{self.class} must implement this method"
    end

    def defend
      raise NotImplementedError, "#{self.class} must implement this method"
    end

      protected_methods

    def set_health(health)
      @health = health
    end

    def get_intelligence
      @intelligence
    end

    def get_strength
      @strength
    end

    def set_pos(row, col)
      @row = row
      @col = col
    end

    def got_wounded
      @health -=1
    end
  end
end

File: Java/src/irrgarten/LabyrinthCharacter.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author javie
 */
public abstract class LabyrinthCharacter {
    String name;
    float intelligence;
    float strength;
    float health;
    int row;
    int col;
    private boolean aux;

    public LabyrinthCharacter(String name, float intelligence, float strength, float health) {
        this.name = name;
        this.intelligence = intelligence;
        this.strength = strength;
        this.health = health;
    }

    public LabyrinthCharacter(LabyrinthCharacter other){
        this.name = other.name;
        this.intelligence = other.intelligence;
        this.strength = other.strength;
        this.health = other.health;
    }
    
    public boolean dead(){
        if(health <= 0){
            aux = true;
        } else {
            aux = false;
        }
        return aux;
    }
    
    public int getRow() {
        return row;
    }

    public int getCol() {
        return col;
    }

    protected float getHealth() {
        return health;
    }

    protected void setHealth(float health) {
        this.health = health;
    }

    protected float getIntelligence() {
        return intelligence;
    }

    protected float getStrength() {
        return strength;
    }
    
    public void setPos(int row, int col){
        this.row = row;
        this.col = col;
    }
    
    public String toString(){
        return "M name: " + this.name + ", Intelligence: " + this.intelligence +
                ", Strength: " + this.strength + ", Health: " + health + ", Row: " + row + ", Col: " + col;
    }
    
    protected void gotWounded(){
        health--;
    }
    
    abstract public float attack();
    abstract public boolean defend(float attack);
    
}
